doc/changes/latest.rst

```diff
@@ -54,6 +54,7 @@ Enhancements
 Changes
 -------
 
+- :func:`~glm.first_level.experimental_paradigm.check_events` will now throw a warning if some events have a 0 second duration and will throw an error if an event has ``NaN`` onset or duration (:gh:`3943` by `Rémi Gau`_).
```

Comment:
:func:`~glm.first_level.experimental_paradigm.check_events`

and

:func:`~glm.first_level.check_events`

give me the same error.

Am I missing something obvious?

Reply:
`check_events` is not exposed in the docs so there is nothing to link to. I'm not sure if it is supposed to be part of the public API or not. If so, it would need to be added to `doc/modules/glm.rst` first before you can use that role

Reply:
I wouldn't put it in the public API.

Reply:
Then we just refer to it with the backticks without the func role

Reply:
Wait... Just to make sure I understand. 

What is the rationale of saying it is not in the public API but check_design_matrix is when both of them are found this init file? 

https://github.com/nilearn/nilearn/blob/main/nilearn/glm/first_level/__init__.py#L24

Reply:
As the codebase stands I wouldn't take what is in the init file for what is in the public API and supposed to be exposed to users since we know there is heterogeneity in the way "private" functions are written and also, we can have public API of a private module so that would be ok to include in the init. I agree though that given the refactoring we are doing regarding this, check_events should either be in a file named with a leading underscore or, since it is only used in design_matrix.py, it can be moved there and renamed with a leading underscore. I don't see that, following this convention, it would then be a problem if it is imported in a module's init file since it'll be clear from the import statement, which would then have a leading underscore, that it is private. This is what I understand from [PEP8](https://peps.python.org/pep-0008/#public-and-internal-interfaces), though we may also want to make a decision on that aspect if it's confusing.

Reply:
OK thanks for the clarification.

OK so for the moment it really feels the public API is whatever the doc days is in the public API. Hopefully things should get clearer once we clean things up.

For now I will use the backticks without the func role.